src/create-with-theme.js

```diff
@@ -1,31 +1,26 @@
 const React = require('react');
-const PropTypes = require('prop-types');
 const channel = require('./channel');
+const createThemeListener = require('./create-theme-listener');
 
 function createWithTheme(CHANNEL = channel) {
+  const themeListener = createThemeListener(CHANNEL);
   return Component =>
     class withTheme extends React.Component {
```

Comment:
In Redux it is `Connect(Something)`
react-intl (very popular as well) -  InjectIntl(Something)